code/game/objects/items/weapons/RCD.dm

```diff
@@ -378,7 +403,186 @@ RCD
+
+
+/obj/item/weapon/construction/rld/proc/checkdupes(var/target)
+	. = list()
+	var/turf/checking = get_turf(target)
+	for(var/obj/machinery/light/dupe in checking)
+		if(istype(dupe, /obj/machinery/light))
+			. |= dupe
+
+
+/obj/item/weapon/construction/rld/afterattack(atom/A, mob/user)
+	range_check(A,user)
+	var/turf/start = get_turf(src)
+	switch(mode)
+		if(REMOVE_MODE)
+			if(istype(A, /obj/machinery/light/))
+				if(checkResource(deconcost, user))
+					to_chat(user, "<span class='notice'>You start deconstructing [A]...</span>")
+					user.Beam(A,icon_state="nzcrentrs_power",time=15)
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					if(do_after(user, decondelay, target = A))
+						if(!useResource(deconcost, user)) 
+							return 0
+						activate()
+						qdel(A)
+						return TRUE
+				return FALSE
+		if(LIGHT_MODE)
+			if(iswallturf(A))
+				var/turf/closed/wall/W = A
+				if(checkResource(floorcost, user))
+					to_chat(user, "<span class='notice'>You start building a wall light...</span>")
+					user.Beam(A,icon_state="nzcrentrs_power",time=15)
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					playsound(src.loc, 'sound/effects/light_flicker.ogg', 50, 0)
+					if(do_after(user, floordelay, target = A))
+						if(!istype(W))
+							return FALSE
+						var/list/candidates = list()
+						var/turf/open/winner = null
+						var/winning_dist = null
+						for(var/direction in GLOB.cardinal)
+							var/turf/C = get_step(W, direction)
+							var/list/dupes = checkdupes(C)
+							if(start.CanAtmosPass(C) && !dupes.len)
+								candidates += C
+						if(!candidates.len)
+							to_chat(user, "<span class='warning'>Valid target not found...</span>")
+							playsound(src.loc, 'sound/misc/compiler-failure.ogg', 30, 1)
+							return FALSE
+						for(var/turf/open/O in candidates)
+							if(istype(O))
+								var/x0 = O.x
+								var/y0 = O.y
+								var/contender = cheap_hypotenuse(start.x, start.y, x0, y0)
+								if(!winner)
+									winner = O
+									winning_dist = contender
+								else
+									if(contender < winning_dist) // lower is better
+										winner = O
+										winning_dist = contender
+						activate()
+						if(!useResource(wallcost, user))
+							return FALSE
+						var/light = get_turf(winner)
+						var/align = get_dir(winner, A)
+						var/obj/machinery/light/L = new /obj/machinery/light(light)
+						L.dir = align
+						L.color = color_choice
+						L.light_color = L.color
+						return TRUE
+				return FALSE
+
+			if(isfloorturf(A))
+				var/turf/open/floor/F = A
+				if(checkResource(floorcost, user))
+					to_chat(user, "<span class='notice'>You start building a floor light...</span>")
+					user.Beam(A,icon_state="nzcrentrs_power",time=15)
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					playsound(src.loc, 'sound/effects/light_flicker.ogg', 50, 1)
+					if(do_after(user, floordelay, target = A))
+						if(!istype(F)) return 0
```

Comment:
put the return on a new line below hte if, same goes for anywhere else we use this pattern, it's a violation of our style guidelines